test/qos1.js

```diff
@@ -509,6 +509,71 @@ test('resend publish on non-clean reconnect QoS 1', function (t) {
   })
 })
 
+test('resend many publish on non-clean reconnect QoS 1', function (t) {
+  t.plan(38)
+
+  const broker = aedes()
+  t.tearDown(broker.close.bind(broker))
+
+  const opts = { clean: false, clientId: 'abcde' }
+  let subscriber = connect(setup(broker), opts)
+
+  subscribe(t, subscriber, 'hello', 1, function () {
+    subscriber.inStream.end()
+
+    const publisher = connect(setup(broker))
+
+    let count = 0
+    while (++count <= 17) {
```

Comment:
The count here needs to be 17 (or higher) to ensure we're above the [`through2` highWaterMark](https://github.com/rvagg/through2/blob/master/through2.js#L70)